c10/core/impl/alloc_cpu.cpp

```diff
@@ -41,6 +41,38 @@ void memset_junk(void* data, size_t num) {
   }
 }
 
+static inline bool is_thp_alloc_enabled() {
+  static bool value = [&](const char* pt) {
+    if (pt != NULL) {
+      return std::atoi(pt);
+    } else {
+      return 0;
+    }
+  }(std::getenv("THP_MEM_ALLOC_ENABLE"));
```

Comment:
This indeed works fine, but wouldn't it be nicer to just move this into the lambda body?
```suggestion
  }(std::getenv("THP_MEM_ALLOC_ENABLE"));
```